plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/RetryTemplateService.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.orchestrator.plugin.flexibleengine.monitor.utils;
+
+import com.huaweicloud.sdk.ces.v1.model.BatchListMetricDataResponse;
+import com.huaweicloud.sdk.ces.v1.model.ListMetricsResponse;
+import com.huaweicloud.sdk.ces.v1.model.ShowMetricDataResponse;
+import com.huaweicloud.sdk.core.internal.model.KeystoneListProjectsResponse;
+import com.huaweicloud.sdk.core.internal.model.Project;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.Header;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * RestTemplate with Retry.
+ */
+@Slf4j
+@Component
+@EnableRetry
+public class RetryTemplateService {
+
+    private static final int RETRY_TIMES = 5;
+    private static final long DELAY_MILLISECONDS = 2000L;
+    private static final double DELAY_MULTIPLIER = 1.5D;
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    /**
+     * Query project info using RestTemplate with Spring-Retry.
+     *
+     * @param httpRequestBase httpRequestBase
+     * @return Project
+     */
+    @Retryable(retryFor = RestClientException.class, maxAttempts = RETRY_TIMES,
```

Comment:
can we also add a `RetryListener` which logs when error happens and retry is being performed?